flutter_local_notifications/lib/src/flutter_local_notifications_plugin.dart

```diff
@@ -62,6 +64,23 @@ class FlutterLocalNotificationsPlugin {
     return true;
   }
 
+  Future<bool> requestPermissions({
```

Comment:
what I meant [here](https://github.com/MaikuB/flutter_local_notifications/pull/311#issuecomment-585354185) was to not have any platform-specific methods in the cross-platform facing API i.e. in this class. Since this plugin now follows the [federated plugin approach](flutter.dev/go/federated-plugins), developers can access the platform-specific implementation of a plugin to access platform-specific functionality if needed.

Originally, I was thinking to have developers following the code similar to what is written below

```dart
 (FlutterLocalNotificationsPlatform.instance
             as IOSFlutterLocalNotificationsPlugin)
         ?.requestPermissions(...)
```

However, it's a bit sub-optimal as it requires developers to import the `flutter_local_notifications_platform_interface` package that is a transitive dependency some might not be aware of. As such my thoughts are to have a static `instance` property within the `IOSFlutterLocalNotificationsPlugin` and `AndroidFlutterLocalNotificationsPlugin` that points to `FlutterLocalNotificationsPlatform.instance` and cast it as the appropriate type. Can you make that change in this PR?

Reply:
Sure, will add ios and android specific `instance` property to plugin class